### ISSUE TITLE
Interactor question

### ISSUE THREAD

**Issue by amahairides** [infinum/iOS-VIPER-Xcode-Templates#29]
Hi, after completing my first Viper project (and creating everything by hand) I decided to try your templates for the next project.
I have a question regarding the Interactor in your templates. Why is there no connection to a presenter? How can the Interactor respond to the presenter (Interactor output). I saw in your examples you are using callbacks. Is this how you imagined it in this template? Or am I missing something?
tl;dr why does the Interactor not have access to a weak presenter property for his output?

**Comment by Truba** [comment]
Hi @amahairides ,

>  I saw in your examples you are using callbacks. Is this how you imagined it in this template?

Yes you are correct, this is exactly how we imagined it. 😄
Nowadays we are mostly using Rx and communicating with `Observables` (or one of the thraits).
But callbacks or any other async tolls like Futures and Promises should do the job.

We did it to reduce the number dependencies in the graph and make everything a bit more simpler. 
Of Course if you prefer to have presenter as a dependency in interactor, you are free to edit the templates locally. It shouldn't take you more that a few minutes.
You would have to update `Interfaces`, `Interactor` and `Wireframe` in:
https://github.com/infinum/iOS-VIPER-Xcode-Templates/tree/master/Templates/Resources/Templates
and then run `main.rb` script to generate templates.

Hope this helps,
:)

**Comment by amahairides** [comment]
Thanks for the quick response! I'll check it out.
